src/main/java/io/jenkins/plugins/opentelemetry/job/step/GitCheckoutStepHandler.java

```diff
@@ -101,14 +102,22 @@ public SpanBuilder createSpanBuilder(@NonNull FlowNode node, @NonNull WorkflowRu
             }
 
             List<Map<String, ?>> extensions = (List<Map<String, ?>>) scm.get("extensions");
-            final Map<String, ?> cloneOption = Iterables.getFirst(extensions, null);
+            if (extensions == null) {
+                LOGGER.log(Level.FINE, "Extensions is null for node {0}, using empty list", node.getId());
+                extensions = Collections.emptyList();
+            }
+                final Map<String, ?> cloneOption = Iterables.getFirst(extensions, null);
```

Comment:
Looks like this was indented unnecessarily:

```suggestion
            final Map<String, ?> cloneOption = Iterables.getFirst(extensions, null);
```

Reply:
please see this -https://github.com/Amankkc/opentelemetry-plugin/pull/1/commits/b93197322e4f0ca733b8b3b0b46172eb22cfc47a